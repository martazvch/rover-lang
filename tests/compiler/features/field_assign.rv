struct Person {
    name: str,
}

fn main() {
    var tom = Person{name = "Tom"}
    tom.name = "Martin"
}

