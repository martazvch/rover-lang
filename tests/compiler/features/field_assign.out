-- main --
get_global index 0, <structure Person>
constant index 0, value Tom
struct_lit index 1
constant index 1, value Martin
get_local index 1
OP_GET_FIELD_REG index 0
OP_REG_ASSIGN
ret_naked

-- global scope --
get_global index 1, <function main>
call index 0

